include/openssl/pkcs12.h

```diff
@@ -55,7 +55,7 @@ typedef struct pkcs12_bag_st PKCS12_BAGS;
 
 /* Compatibility macros */
 
-#if OPENSSL_API_COMPAT < 0x10100000L
+#ifndef OPENSSL_API_1_1_0
```

Comment:
Do all the header files where you make similar changes ultimately include opensslconf.h? If not applications that define OPENSSL_API_COMPAT may not pick up the definitions of OPENSSL_API_1_1_0 etc and hence start declaring things that shouldn't be declared.

Reply:
This is a good point, and hard to detect as written.  So maybe those macros should be turned into booleans instead, and checked with `#if` rather than `#ifdef`, that would make for better detection.

Reply:
I'm not sure I understand how that helps?

Reply:
By making these macros boolean, we *demand* that they be defined, which *demands* that `opensslconf.h` be included first.  Our own builds will fail if this isn't the case...  if you want this checked even more thoroughly, it wouldn't be too hard to generate a build test program with all deprecated symbols and test that it builds properly.